File: 0902/무지의 먹방 라이브.py
from collections import defaultdict
def solution(food_times, k):
    
    
    answer = 0
    N = len(food_times)
    s = sum(food_times)
    
    # 중단될때까지 먹을 수 없는 경우
    if s <= k:
        return -1
    
    i = 0 # 실제 시간 흐름
    j = 0 # 현재 음식의 위치
    
    # k초까지 먹방 진행
    while i < k:
        
        if food_times[j%N] > 0:
            food_times[j%N] -= 1
            
            i += 1
            j += 1
            
        else:
            j += 1
        
    # k+1 번째를 찾기 위해서 진행    
    
    while food_times[j%N] == 0:
        j += 1
        
    # 인덱스 -> 번째로 변환
    answer = (j%N)+1
    
    return answer

File: 0902/무지의 먹방 라이브.swift
import Foundation

func solution(_ food_times:[Int], _ k:Int64) -> Int {
    var dict: [Int64:Int64] = [:]
    let count = food_times.count
    var dicCount: Int64 = 0
    
    for i in 0..<count {
        var j: Int64 = Int64(i)
        while j < count*food_times[i] + i {
            dict[j] = Int64(i)
            j += Int64(count)
            dicCount += 1
        }
    }
    
//    print(dict[k+1])
    var answer = -1
    var keyCount: Int64 = -1
    for keys in dict.keys.sorted() {
        keyCount += 1
        if keyCount == k {
            answer = Int(dict[keys]!)+1
        }
    }
    
    return answer
}

